editor/plugins/texture_editor_plugin.cpp

```diff
@@ -30,132 +30,60 @@
+	texture_display->set_stretch_mode(TextureRect::STRETCH_KEEP_ASPECT_CENTERED);
+	texture_display->set_expand(true);
+	add_child(texture_display);
 
-		int tex_width = texture->get_width() * size.height / texture->get_height();
-		int tex_height = size.height;
-
-		if (tex_width > size.width) {
-			tex_width = size.width;
-			tex_height = texture->get_height() * tex_width / texture->get_width();
-		}
-
-		// Prevent the texture from being unpreviewable after the rescale, so that we can still see something
-		if (tex_height <= 0) {
-			tex_height = 1;
-		}
-		if (tex_width <= 0) {
-			tex_width = 1;
-		}
-
-		int ofs_x = (size.width - tex_width) / 2;
-		int ofs_y = (size.height - tex_height) / 2;
-
-		if (Object::cast_to<CurveTexture>(*texture)) {
-			// In the case of CurveTextures we know they are 1 in height, so fill the preview to see the gradient
-			ofs_y = 0;
-			tex_height = size.height;
-		} else if (Object::cast_to<GradientTexture>(*texture)) {
-			ofs_y = size.height / 4.0;
-			tex_height = size.height / 2.0;
-		}
-
-		draw_texture_rect(texture, Rect2(ofs_x, ofs_y, tex_width, tex_height));
-
-		Ref<Font> font = get_theme_font("font", "Label");
-		int font_size = get_theme_font_size("font_size", "Label");
+	if (p_show_metadata) {
+		Label *metadata_label = memnew(Label);
 
 		String format;
-		if (Object::cast_to<ImageTexture>(*texture)) {
-			format = Image::get_format_name(Object::cast_to<ImageTexture>(*texture)->get_format());
-		} else if (Object::cast_to<StreamTexture2D>(*texture)) {
-			format = Image::get_format_name(Object::cast_to<StreamTexture2D>(*texture)->get_format());
+		if (Object::cast_to<ImageTexture>(*p_texture)) {
+			format = Image::get_format_name(Object::cast_to<ImageTexture>(*p_texture)->get_format());
+		} else if (Object::cast_to<StreamTexture2D>(*p_texture)) {
+			format = Image::get_format_name(Object::cast_to<StreamTexture2D>(*p_texture)->get_format());
 		} else {
-			format = texture->get_class();
-		}
-		String text = itos(texture->get_width()) + "x" + itos(texture->get_height()) + " " + format;
-
-		Size2 rect = font->get_string_size(text, font_size);
-
-		Vector2 draw_from = size - rect + Size2(-2, font->get_ascent(font_size) - 2);
-		if (draw_from.x < 0) {
-			draw_from.x = 0;
+			format = p_texture->get_class();
 		}
 
-		draw_string(font, draw_from + Vector2(2, 2), text, HALIGN_LEFT, size.width, font_size, Color(0, 0, 0, 0.5));
-		draw_string(font, draw_from - Vector2(2, 2), text, HALIGN_LEFT, size.width, font_size, Color(0, 0, 0, 0.5));
-		draw_string(font, draw_from, text, HALIGN_LEFT, size.width, font_size, Color(1, 1, 1, 1));
-	}
-}
+		metadata_label->set_text(itos(p_texture->get_width()) + "x" + itos(p_texture->get_height()) + " " + format);
 
-void TextureEditor::_texture_changed() {
-	if (!is_visible()) {
-		return;
-	}
-	update();
-}
+		Ref<Font> metadata_label_font = theme->get_font("expression", "EditorFonts");
+		metadata_label->add_theme_font_size_override("font_size", 32);
```

Comment:
Scale for hiDPI displays:

```suggestion
		metadata_label->add_theme_font_size_override("font_size", 32 * EDSCALE);
```

Reply:
@Calinou fixed.